google-cloud-spanner/src/main/java/com/google/cloud/spanner/AsyncTransactionManagerImpl.java

```diff
@@ -174,10 +176,6 @@ public void onSuccess(Timestamp result) {
 
   @Override
   public TransactionContextFuture resetForRetryAsync() {
-    if (txn == null || (!txn.isAborted() && txnState != TransactionState.ABORTED)) {
```

Comment:
This check is no longer needed here as it is checked in the `SessionPoolAsyncTransactionManager`.